src/mam4xx/mam4_amicphys.hpp

```diff
@@ -0,0 +1,2371 @@
+  //
+  Real dgn_a[num_modes] = {};
+  Real dgn_awet[num_modes] = {};
+  Real wetdens[num_modes] = {};
+  assign_1d_array(num_modes, zero,         // in
+                  dgn_a);                  // out
+  assign_1d_array(num_modes, zero,         // in
+                  dgn_awet);               // out
+  assign_1d_array(num_modes, one_thousand, // in
+                  wetdens);                // out
+
+  for (int imode = 0; imode < num_modes; ++imode) {
+    dgn_a[imode] = dgncur_a[imode];
+    dgn_awet[imode] = dgncur_awet[imode];
+    wetdens[imode] = haero::max(one_thousand, wetdens_host[imode]);
+  }
+
+  Real qsub_tendaa[gas_pcnst][nqtendaa()][maxsubarea()] = {};
+  Real qqcwsub_tendaa[gas_pcnst][nqqcwtendaa()][maxsubarea()] = {};
+  mam_amicphys_1gridcell(
+      // in
+      config, deltat, nsubarea, ncldy_subarea, iscldy_subarea, afracsub, temp,
+      pmid, pdel, zm, pblh, relhumsub, dgn_a, dgn_awet, wetdens, qsub1, qsub2,
+      qqcwsub2, qsub3, qqcwsub3,
+      // inout
+      qaerwatsub3, qsub4, qqcwsub4, qaerwatsub4, qsub_tendaa, qqcwsub_tendaa);
+
+  //=================================================================================================
+  // Aerosol microphysics calculations done for all subareas. Form new grid cell
+  // mean mixing ratios.
+  //=================================================================================================
+  // Gases and aerosols
+  //----------------------
+  // Calculate new grid cell mean values
+  Real qgcm4[gas_pcnst];
+  Real qqcwgcm4[gas_pcnst];
+
+  form_gcm_of_gases_and_aerosols_from_subareas(
+      // in
+      nsubarea, ncldy_subarea, afracsub, qsub4, qqcwsub4, qqcwgcm3,
+      // out
+      qgcm4, qqcwgcm4);
+
+  // Copy grid cell mean values to output arrays
+  for (int icnst = 0; icnst < gas_pcnst; ++icnst) {
+    if (lmapcc_all(icnst) > 0) {
+      qq[icnst] = qgcm4[icnst];
+    }
+    if (lmapcc_all(icnst) >= lmapcc_val_aer()) {
+      qqcw[icnst] = qqcwgcm4[icnst];
+    }
+  }
+
+  //================================================================
+  // Process diagnostics of the current grid cell
+  //================================================================
+  Real qgcm_tendaa[gas_pcnst][nqtendaa()];
+  Real qqcwgcm_tendaa[gas_pcnst][nqqcwtendaa()];
+  get_gcm_tend_diags_from_subareas(
+      // in
+      nsubarea, ncldy_subarea, afracsub, qsub_tendaa, qqcwsub_tendaa,
+      // out
+      qgcm_tendaa, qqcwgcm_tendaa);
+
+#if 0
+  //This code is for diagnostics only
+  // Get gravity
+  using C                      = physics::Constants<Real>;
+  static constexpr auto gravit = C::gravit;  // Gravity [m/s2]
+
+accumulate_column_tend_integrals( pdel, gravit,                         // in
+                                        qgcm_tendaa,         qqcwgcm_tendaa,         // in
+                                        q_coltendaa(ii,:,:), qqcw_coltendaa(ii,:,:)  )// inout
+
+ncluster_3dtend_nnuc(ii,kk) = misc_vars_aa%ncluster_tend_nnuc_1grid
+
+#endif
```

Comment:
Should this be kept around?

Reply:
We can remove code from `#if 0` to `#endif`